src/commands/games/Slots.ts

```diff
@@ -119,33 +244,126 @@ export class Slots extends Command
         const grid = parseInt(int.options.get("slots_grid")?.value?.toString() ?? "3");
 
         if (coins < bet) {
-            return await this.reply(int, embedify(`Insufficient coins\nYour balance is: ${coins}`));
+            return await this.editReply(int, embedify(`Insufficient coins\nYour balance is: ${coins}`));
         }
 
-        await this.reply(int, "Shuffling slots...");
+        await this.editReply(int, "Shuffling slots...");
 
-        let resultStr = "";
-        const result = [];
-        const emojis = Array.from(this.SLOTS.keys());
+        const chance = Math.random();
+        let win: Slot | null = null;
 
-        for (let i = 0; i < grid * grid; i++) {
-            if (i % grid == 0) {
-                resultStr += "\n";
+        // TODO: a chance of getting your money back (maybe 5%?)
+
+        for (let i = 0; i < this.SLOTS.length; i++) {
+            if (chance < this.SLOTS[i].chance.get(grid)!) {
+                win = this.SLOTS[i];
+                break;
             }
-            const rand = emojis[Math.floor(Math.random() * emojis.length)];
-            resultStr += rand;
-            result.push(rand);
         }
 
-        await this.editReply(int, resultStr);
-
-        const win = this.checkWin(result, grid);
+        let resultStr = "";
+        const emojis = this.SLOTS.map((slot) => {
+            return slot.emoji;
+        });
 
         if (win) {
```

Comment:
couldn't this potentially generate two winning patterns at once?

Reply:
it can, but idc :low_quality_pepe:
(fuck i have to actually fix this shit)

Reply:
in that case why not just roll with it and just recount the final score

Reply:
because that's not how i want it to be + which emote takes precedence or do i count both?

Reply:
both, it would be the jackpot to have a 5x5 of eggplant  
actual billionaire jeef benzos simulator

Reply:
hmmmm too lazy. either you or canny do that :low_quality_pepe:

Reply:
understandable

Reply:
i did it anyway. plz revrev